src/quota_aggregator_impl.cc

```diff
@@ -210,25 +228,16 @@ ::google::protobuf::util::Status QuotaAggregatorImpl::FlushAll() {
 
 // OnCacheEntryDelete will be called behind the cache_mutex_
 // no need to consider locking at this point
-//
-// if the element is refreshing and aggregated while waiting for the
-// response, tokens should be aggregated
 void QuotaAggregatorImpl::OnCacheEntryDelete(CacheElem* elem) {
-  if (elem->in_flight() && in_flush_all_ == false) {
-    // keep the cached entry remain in the cache if the entry is waiting for
-    // the response
-    cache_->Insert(elem->signature(), elem, 1);
-    return;
-  }
-
-  if (elem->is_aggregated() == true && in_flush_all_ == false) {
-    // Insert the element back to the cache while aggregator is waiting for the
-    // response.
-    elem->set_in_flight(true);
+  if (in_flush_all_ == false) {
+    // insert the element back to the cache
     cache_->Insert(elem->signature(), elem, 1);
 
-    AddRemovedItem(elem->ReturnAllocateQuotaRequestAndClear(
+    if (elem->in_flight() == false && elem->is_aggregated()) {
+      elem->set_in_flight(true);
+      AddRemovedItem(elem->ReturnAllocateQuotaRequestAndClear(
```

Comment:
last_refresh_time should be set here.  at request sending time, not response time.